src/Operator/Operators.php

```diff
@@ -0,0 +1,93 @@
+<?php
+
+/*
+ * This file is part of Twig.
+ *
+ * (c) Fabien Potencier
+ *
+ * For the full copyright and license information, please view the LICENSE
+ * file that was distributed with this source code.
+ */
+
+namespace Twig\Operator;
+
+use Twig\Operator\Binary\AbstractBinaryOperator;
+use Twig\Operator\Unary\AbstractUnaryOperator;
+
+class Operators implements \IteratorAggregate
+{
+    private array $operators = [];
+    private array $aliases = [];
+    private ?\WeakMap $precedenceChanges = null;
+
+    public function __construct(
+        array $operators = [],
+    ) {
+        $this->add($operators);
+    }
+
+    /**
+     * @param array<AbstractOperator> $operators
+     *
+     * @return $this
+     */
+    public function add(array $operators): self
+    {
+        $this->precedenceChanges = null;
+        foreach ($operators as $operator) {
+            $this->operators[$operator->getArity()->value][$operator->getOperator()] = $operator;
+            foreach ($operator->getAliases() as $alias) {
+                $this->aliases[$operator->getArity()->value][$alias] = $operator;
+            }
+        }
+
+        return $this;
+    }
+
+    public function getUnary(string $name): ?AbstractUnaryOperator
+    {
+        return $this->operators[OperatorArity::Unary->value][$name] ?? ($this->aliases[OperatorArity::Unary->value][$name] ?? null);
+    }
+
+    public function getBinary(string $name): ?AbstractBinaryOperator
```

Comment:
this getter expects that a binary operator extends AbstractBinaryOperator, but there is no such validation when adding them. Neither the ExtensionInterface nor the OperatorInterface explain that requirement.